include/odp/api/spec/packet.h

```diff
@@ -2210,6 +2258,18 @@ void odp_packet_vector_print(odp_packet_vector_t pktv);
  */
 uint64_t odp_packet_vector_to_u64(odp_packet_vector_t hdl);
 
+/*
+ * Check reassembly status of the packet
+ *
+ * When reassembly is enabled, application must check reassembly status of
+ * the packet before performing any packet operation.
+ *
+ * @param  pkt Packet handle
+ * @return     Reassembly status
+ *
+ */
+odp_packet_reassem_status_t odp_packet_reassem_status(odp_packet_t pkt);
```

Comment:
General comment: "reassem" is not much shorter than "reassembly". I would suggest "reass" (or "reassembly").

> Application must check reassembly status of the packet before performing any packet operation

I am not sure if this is accurate. An application must not attempt to do packet operation for incomplete reassembly results (but must free the packet or call recover_frags / reass_result). For ensuring that the application may have to call the status function but calling the status function does not turn the incomplete packet into something that can be processed normally.

Or was the idea here that even for completed reassembly the processing would be incomplete before the status function gets called, like was proposed for IPsec post-processing?

Reply:
Your understanding is correct. recover_frags/reass_result() is the actual mandatory one. Idea is to not loose individual fragments. I'll reword to reflect that reass_result() would be the mandatory call for INCOMPLETE cases.

Reply:
@psavol @JannePeltonen Since we are renaming recover_frags to reass_result(), I was wondering whether we should combine the two(status & recover) into a single API (reass_result)